aws-predictions/src/main/java/com/amplifyframework/predictions/aws/service/AWSRekognitionService.java

```diff
@@ -226,6 +246,60 @@ void recognizeCelebrities(
         return celebrities;
     }
 
+    private IdentifyTextResult detectPlainText(Image image) throws PredictionsException {
+        DetectTextRequest request = new DetectTextRequest()
+                .withImage(image);
+
+        // Read text in the given image via Amazon Rekognition
+        final DetectTextResult result;
+        try {
+            result = rekognition.detectText(request);
+        } catch (AmazonClientException serviceException) {
+            throw new PredictionsException(
+                    "Amazon Rekognition encountered an error while detecting text.",
+                    serviceException, "See attached service exception for more details."
+            );
+        }
+
+        StringBuilder fullTextBuilder = new StringBuilder();
```

Comment:
I like the pattern you used elsewhere where split out the "convert to result type" step into a helper.

Reply:
`AWSTextractService#processTextractBlocks()` was split into its own method mainly because the same logic was required for both `detectDocumentText()` and `analyzeDocument()` methods. Generally for other services, I included the conversion directly inside the method calling the service APIs.